core/config.go

```diff
@@ -85,6 +86,15 @@ func (d *Config) ListenAddress(defaultAddr string) string {
 	return defaultAddr
 }
 
+// ListenPort returns the given default port or the listen port stored
+// in the config thanks to WithListenPort
+func (d *Config) ListenPort(defaultPort string) string {
```

Comment:
Should not need to put the default port in argument here. You can place the constant in this file (see the other constant at the beginning) and just put this constant by default when you create the config, and override it when call `WithListenPort`. See other variables for example on how to do this